crates/cli-support/src/js/mod.rs

```diff
@@ -475,9 +475,20 @@ impl<'a> Context<'a> {
                     ",
                 );
 
-                if needs_manual_start {
-                    start = Some("\nwasm.__wbindgen_start();\n".to_string());
-                }
+                start = {
+                    let mut start = String::new();
+                    if matches!(self.config.mode, OutputMode::Node { module: true }) {
+                        start.push_str(&self.generate_node_imports());
+                        start.push_str(&self.generate_node_wasm_loading(Path::new(&format!(
+                            "./{}_bg.wasm",
+                            module_name
+                        ))));
+                    }
+                    if needs_manual_start {
+                        start.push_str("\nwasm.__wbindgen_start();\n");
+                    }
+                    (!start.is_empty()).then_some(start)
+                };
```

Comment:
Apologies, this is what I was proposing:
```suggestion
                if matches!(self.config.mode, OutputMode::Node { module: true }) {
                    let start = start.get_or_insert_with(String::new);
                    start.push_str(&self.generate_node_imports());
                    start.push_str(&self.generate_node_wasm_loading(Path::new(&format!(
                        "./{}_bg.wasm",
                        module_name
                    ))));
                }
                if needs_manual_start {
                    start
                        .get_or_insert_with(String::new)
                        .push_str("\nwasm.__wbindgen_start();\n");
                }
```

Reply:
That works too, I guess it saves an allocation in case neither matches. 😄

Reply:
FWIW, I just found it easier to reason about the code.
[`String::new()`](https://doc.rust-lang.org/1.80.0/std/string/struct.String.html#method.new) doesn't actually allocate.

Reply:
Yeah I can see it.